src/jit/PerfDump.cpp

```diff
@@ -66,16 +70,36 @@ PerfDump& PerfDump::instance()
 PerfDump::PerfDump()
     : m_pid((uint32_t)getpid())
     , m_codeLoadIndex(0)
+    , m_line(0)
 {
-    std::string tmpName = "./jit-" + std::to_string(m_pid) + ".dump";
+    const char* path = getenv("WALRUS_PERF_DIR");
+
+    if (path == nullptr || *path == '\0') {
+        return;
+    }
+
+    m_sourceFileName = std::string(path) + "/jit-" + std::to_string(m_pid) + "-codedump.txt";
```

Comment:
What is the purpose of `m_sourceFileName`?
And it seems that this file is opened, but not closed after execution.

Reply:
True, I will fix that. The purpose is generating a text dump from the Walrus byte code, and this "source" file is referred by the debug line info. In Walrus byte code, we cannot connect the original source in any way, but this way we still get info where the execution is mostly happens.

Example dump:
```
--- FUNCTION:5 [stackAlloc] ---
Prolog
[0] MoveI32: R0(r0)
[1] Const32: R0(imm)
[2] GlobalGet32: R0(r1)
[3] I32Sub: P0(r1), P1(r0), R0(r1)
[4] I32And: P0(r1), P1(imm), R0(r1)
[5] MoveI32: P0(r1), R0(r0)
[6] GlobalSet32: P0(r1), T0(r3)
[7] End: P0(r0)
Epilog
```
Example mixed code:
```
Percent│      [550] MoveF32: P0(r0), R0(o:152)                                                                            ▒
  0.16 │        movss    %xmm0,0x98(%r15)                                                                                 ▒
       │      [551] F32Store: P0(r9), P1(r0), T0(r1), T1(r2)                                                              ▒
       │        mov      %r13d,%esi                                                                                       ▒
       │        mov      0x38(%rbx),%rax                                                                                  ▒
       │        add      $0x84,%rsi                                                                                       ▒
       │        cmp      $0xfffffc,%rsi                                                                                   ▒
       │      → ja       3067                                                                                             ▒
       │        movss    %xmm0,(%rax,%rsi,1)                                                                              ▒
       │      [552] F32Load: P0(r9), R0(o:144), T0(r1), T1(r2)                                                            ▒
       │        mov      %r13d,%esi                                                                                       ▒
       │        mov      0x38(%rbx),%rax                                                                                  ▒
       │        add      $0x6c,%rsi                                                                                       ▒
       │        cmp      $0xfffffc,%rsi                                                                                   ▒
       │      → ja       3067                                                                                             ▒
  0.32 │        movss    (%rax,%rsi,1),%xmm4                                                                              ▒
       │        movss    %xmm4,0x90(%r15)                                                                                 ▒
       │      [553] MoveF32: P0(o:168), R0(o:288)                                                                         ▒
       │        movss    0xa8(%r15),%xmm4                                                                                 ◆
       │        movss    %xmm4,0x120(%r15)                                                                                ▒
       │      [554] F32Sub: P0(o:232), P1(o:212), R0(o:296)                                                               ▒
  0.48 │        movss    0xe8(%r15),%xmm4                                                                                 ▒
       │        subss    0xd4(%r15),%xmm4                                                                                 ▒
       │        movss    %xmm4,0x128(%r15)                     
```

Reply:
So, creating and using this 'source' file, we can analyze performance bottleneck of JIT code through perf tool, am I right?

Reply:
Yes. Without any source information, it is hard to figure out which machine code is generated from which byte code. This way at least we see the WebAssembly (Walrus to be more precise) byte code.